### PR TITLE
Accessibility: Text Set Labels

### LINKED ISSUE #6517: Text Set Labels
## Feature Description

In the process of fixing text sets in the editor to be navigable by keyboard arrows, I was testing with a screen reader and realized we don't have any kind of labeling for text sets. Right now the text read is just `Insert Text Set` - there's no description of what it looks like. 

I'm curious if we want to add labeling to text sets. If yes, then we need that list before development can start. If no, then we can either leave the aria label as is or update it accordingly. 

## Additional Context

Perhaps some kind of label from the cross section of text set type and font as in figma? 

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

Following the below thread of comments we will do the following: 
- Label by text set section (ie List or Editorial) with number it is in the view starting with 1 and increasing as long as the text set is under the same grouping. 
- Labels are hidden and should be announced via screen reader and present in accessibility of dom.

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Context

Text sets were missing aria labels :(  

## Summary

Update text set json to store an array called `fonts` that has all fonts text set uses so that we can create a label for text sets that is descriptive. Store fonts array as well as text set category with elements, use those two things in `textSetsPane` to create a label, pass label to text set list item.

## Relevant Technical Choices

@choumx - We had discussed setting labels as `[Text Set Type] #` in the view but since that was having me modify the data structure anyways I thought we might as well include the fonts in use - I think it makes the list items more descriptive outside of visuals.  Let me know if you're ok with that?

## User-facing changes

Text sets will now have an aria label that is actually descriptive. No visual changes, but if you go through text sets with a screen reader or look at the accessibility tag of a text set list item there is a difference.

It'll be structured as `[text set type] with [fonts present in set] font(s)`, so like this: `cover text set with Roboto font` or `cover text set with Source Serif Pro, Lato fonts`. 

## Testing Instructions

Navigate to text sets in the editor and turn on a screen reader, each text set should be announced with its text set type as well as fonts used in it.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6517

### CODE REVIEW

File: assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js [hunk 1]
```diff
@@ -92,9 +92,23 @@ function TextSetsPane({ paneRef }) {
...
+              'Text set %1$s with %2$s fonts',
+              textSetFonts.length,
+              'web-stories'
+            ),
+            textSetCategory,
+            textSetFonts.join(
+              /* translators: delimiter used in a list */
+              __(', ', 'web-stories')
+            )
+          ),
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js]: @swissspidy - can I get your eyes on this i18n set up if you have a second to verify that it's right?

Comment (swissspidy) [reply assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js]: The translator comment says "text set category" but the string actually contains a list of fonts (`%2$s font` / `%2$s fonts`). So I'd update the translator comment to `/* translators: 1: text set title. 2: list of fonts. */`

**Review by samwhale — APPROVED** [review]
NiceEEe!

File: assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js (hunk 1 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js]: gosh dang it i was so focused on the translation i forgot to proof read. thanks 🤦‍♀️

File: assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js [hunk 2]
```diff
@@ -92,9 +92,23 @@ function TextSetsPane({ paneRef }) {
...
           id: `text_set_${uuidv4()}`,
+          title: sprintf(
+            /* translators: 1: text set category. 2: list of fonts. */
+            _n(
+              'Text set %1$s with %2$s font',
+              'Text set %1$s with %2$s fonts',
+              textSetFonts.length,
+              'web-stories'
+            ),
+            textSetCategory,
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js]: What are some example values for `textSetCategory`?

Comment (BrittanyIRL) [reply assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js]: Screen Shot 2021-04-14 at 11 08 42 AM

Comment (BrittanyIRL) [reply assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js]: oooo let me map those to the constants so they translate

**Review by embarks — APPROVED** [review]
https://user-images.githubusercontent.com/41136059/114766084-ced97380-9d1a-11eb-8cef-062474142f41.mp4

Works for me!

File: assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js [hunk 3]
```diff
@@ -92,9 +92,23 @@ function TextSetsPane({ paneRef }) {
 
   const addIdToTextSets = useCallback(
     (textSetsArray = []) =>
-      textSetsArray.map((elements) => {
+      textSetsArray.map(({ textSetFonts = [], textSetCategory, elements }) => {
         return {
           id: `text_set_${uuidv4()}`,
+          title: sprintf(
```
Comment (dreamofabear) [inline assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js:98]: Nit: I'd imagine a field named `title` would be visible. `ariaLabel`?

**Review by dreamofabear — APPROVED** [review]
Nice!

File: assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js (hunk 3 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js:98]: yeah i thought about it, we're using `title` on page templates for this same structure though and it's really a title of the text set that doesn't need to just be the ariaLabel. I don't really care one way or other if you feel strongly about this rename.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**